### ISSUE TITLE
Weird MultipleObjectsReturned for Social Apps under get_app function

### ISSUE THREAD

**Issue by simkimsia** [pennersr/django-allauth#3495]
![image](https://github.com/pennersr/django-allauth/assets/245021/d518c7ac-6f4a-4200-8feb-9075b87075ed)

<img width="865" alt="image" src="https://github.com/pennersr/django-allauth/assets/245021/80515644-3e84-4921-b981-616291be0f2d">

![settings](https://github.com/pennersr/django-allauth/assets/245021/b9eb6049-32a5-4922-b774-5244f966024e)

I am using django 4.2.6
allauth 0.57

and just subscribed to the 128 dollar monthly tier

**Comment by simkimsia** [comment]
Now using vscode's attach to running container and ipdb to debug

![image](https://github.com/pennersr/django-allauth/assets/245021/acf10544-f7a2-4bd2-82dd-bc4527960ac8)
at the list_apps the db_apps correctly fetch just one app

Around here, it will add the <SocialApp > that's not in the database

![image](https://github.com/pennersr/django-allauth/assets/245021/ed3212a0-9a0f-4dfa-92dd-7dacd7b88816)

According to this

<img width="598" alt="image" src="https://github.com/pennersr/django-allauth/assets/245021/f4a7b3db-9588-499b-9568-04d1890a045f">
It's meant to return a blended list of db/settings

So am I supposed to only use one or the other and not both?

**Comment by simkimsia** [comment]
Oh i get it now. i am supposed to use only one or the other and not both, right?

**Comment by pennersr** [comment]
You either configure the SocialApp in the database via the Django admin, or, you configure one in your settings. Don't do both, as then you can indeed run into a situation where it is not clear which app the code should be picking.

PS: Thanks for sponsoring!

**Comment by simkimsia** [comment]
I am in two minds about closing this ticket.

I feel ready to close but on the other hand I like to ask if tehre's a place in the documentation where I can highlight this to avoid this facepalm moment for future users.

Can advise?

**Comment by pennersr** [comment]
Good point -- let's keep this open to improve the docs indeed.

**Comment by simkimsia** [comment]
would you mind directing me where in the docs i should write about this? happy to contribute back

**Comment by aleksaterzic** [comment]
I have encountered the same with my Apple provider and figured out that now, you can't do both. 
There's still a small issue (at least for me).

Apple asks for `certificate_key`, which differs from other providers like Google or Facebook and it seems that I cannot really use the DB way for this, right? (maybe the new `settings` field can be utilized for this?)

So now, I have to clean up all of the entries in `socialapp`, `socialapp_sites` and `socialtoken` for apple so I can use the configuration from settings.
I've tested it, it doesn't really break anything, but still I don't feel it's the right way to do it

**Comment by pennersr** [comment]
@aleksaterzic Good point. The Apple `certificate_key` was not properly supporting the new `app.settings` based approach yet, but that has been rectified here now: 104416b1

**Comment by pennersr** [comment]
> would you mind directing me where in the docs i should write about this? happy to contribute back

I have done a first stab of documenting things over here: 67c5fbcb -- feel free to file a PR to extend the documentation where necessary.

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3495
